source/slang/slang-check-overload.cpp

```diff
@@ -1671,6 +1702,16 @@ namespace Slang
         {
             if (IsErrorExpr(arg))
                 return CreateErrorExpr(expr);
+
+            // If this argument is itself an overloaded value then we can't
+            // sensibly continue
+            if(as<OverloadedExpr>(arg) || as<OverloadedExpr2>(arg))
```

Comment:
This is unfortunate, but you won't have this limitation if __tc_call is implemented as a high-order expr like __bwd_diff.

Reply:
perhaps. This check is also overly conservative. If this parameter is a monotype, then we can try to unify each overload with it: i.e. resolve this overload with imaginary arguments conjured from the type of the parameter it's being passed as. I'm reluctant to implement this though.